### ISSUE TITLE
Helm bug in localpv with replicated storage

### ISSUE THREAD

**Issue by blmhemu** [openebs/openebs#3765]
## Description
When doing helm install openebs first install localpv and then uninstalls it - this is possibly due to nested usage of localpv-provisioner in both mayastore and openebs.

## Expected Behavior
localpv-provisioner should be installed and deployment present.

## Current Behavior
localpv-provisioner deployment is first installed and then terminated.

## Possible Solution
Just have localpv-provisioner in one place in the helm chart.

## Steps to Reproduce
This is the values config passed.
```
    values: {
      // Disable local engines
      engines: {
        'local': {
          zfs: { enabled: false },
          lvm: { enabled: false },
        },
      },
      mayastor: {
        // Single node
        etcd: { replicaCount: 1 },
        nats: { cluster: { enabled: false } },
        // See https://www.talos.dev/v1.8/kubernetes-guides/configuration/storage/#deploy-mayastor
        csi: { node: { initContainers: { enabled: false } } },
        // Without this, io-engine crashes
        io_engine: { envcontext: 'iova-mode=va' },
        // No bells and whistles
        obs: { callhome: { enabled: false } },
        'loki-stack': { enabled: false },
        // Use xfs as default fs
        storageClass: {
          parameters+: {
            fsType: 'xfs',
          },
        },
      },
    },

```
## Screenshots
See that localpv-provisioner pods are not present.
![image](https://github.com/user-attachments/assets/bcbd8441-6afc-4634-958d-7a78ea23f2e6)

## Your Environment

* `kubectl get nodes`:
* `kubectl get pods --all-namespaces`:
* `kubectl get services`:
* `kubectl get sc`:
* `kubectl get pv`:
* `kubectl get pvc`:
* OS (from `/etc/os-release`):
* Kernel (from `uname -a`):
* Install tools:
* Others:

**Comment by blmhemu** [comment]
A sub complaint - I had to manually add `io_engine: { envcontext: 'iova-mode=va' },` for openebs to work. Without this io_engine crashes.

**Comment by niladrih** [comment]
Hmm. What are exact charts that you've tried to install together here? Could you share the output of `helm ls -n <namespace>`

**Comment by blmhemu** [comment]
Ok here is some interesting observation
```
apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    helm.sh/hook: pre-upgrade
    helm.sh/hook-delete-policy: hook-succeeded
    helm.sh/hook-weight: "0"
  labels:
    app.kubernetes.io/instance: openebs
    app.kubernetes.io/managed-by: Helmraiser
    app.kubernetes.io/version: 4.1.0
    helm.sh/chart: openebs-4.1.0
    tanka.dev/environment: db4ebdd6fded31f5f4ef8a39f60a3d880f75018172dfd95b
  name: openebs-pre-upgrade-hook
  namespace: openebs
spec:
  template:
    metadata:
      labels:
        app.kubernetes.io/instance: openebs
        app.kubernetes.io/managed-by: Helm
        helm.sh/chart: openebs-4.1.0
      name: openebs-pre-upgrade-hook
    spec:
      containers:
      - args:
        - (kubectl annotate --overwrite crd volumesnapshots.snapshot.storage.k8s.io
          volumesnapshotclasses.snapshot.storage.k8s.io volumesnapshotcontents.snapshot.storage.k8s.io
          helm.sh/resource-policy=keep || true) && (kubectl -n openebs delete deploy
          -l openebs.io/component-name=openebs-localpv-provisioner --ignore-not-found)
        command:
        - /bin/sh
        - -c
        image: docker.io/bitnami/kubectl:1.25.15
        imagePullPolicy: IfNotPresent
        name: pre-upgrade-job
      restartPolicy: Never
      serviceAccountName: openebs-pre-upgrade-hook

```

This is the templated output for pre-upgrade hook. Observe how it's deleting the actual localpv deployment `(kubectl -n openebs delete deploy -l openebs.io/component-name=openebs-localpv-provisioner --ignore-not-found)`

**Comment by blmhemu** [comment]
Also, when i install using just the mayastor chart, this does not happen.

**Comment by blmhemu** [comment]
> Ideally, you'd want to use the mayastor from the openebs/openebs chart.
Time to update the docs :)

My setup is still in the infancy so i nuked it and installed from mayastor chart.

**Comment by balaharish7** [comment]
This issue is addressed in the [PR #501](https://github.com/openebs/website/pull/501).
